release/3.2.0

n8rzz opened the pull request:
Resolves #700, #706, #736, #742, #744, #747, #753, #754, #756, #760, #763.

This release will contain a complete re-write of the command parser (currently in PR on my fork).  **This will change how commands are used in the sim**.  

There will be more information forthcoming on command changes as we get closer to the release date.  TL;DR _every_ command will require a space between the command and its arguments.  
ex: `fh270` will become `fh 270`.

This release is targeted to be completed on 12/20/2016.  

- view a list of current, finished changes in the [release/3.2.0 CHANGELOG](https://github.com/n8rzz/atc/blob/release/3.2.0/CHANGELOG.md).
- play around with the current state of v3.2.0 on our [release environment](https://atc-release.herokuapp.com)
- see what else we're planning to include _(subject to change)_ [milestone v3.2.0](https://github.com/n8rzz/atc/issues?q=is%3Aopen+is%3Aissue+milestone%3Av3.2.0)

If you find a bug or figured out a way to break something, please log an [issue](https://github.com/n8rzz/atc/issues) on [my fork](https://github.com/n8rzz/atc/).  If you have any questions about upcoming changes, please log them here.

n8rzz committed:
feature/ATC-53 - Creates getters for currentLeg and currentWaypoint
* Updates refs for new getters
* Begins abstracting waypoint creation to individual methods
* Adds _get for setting properties from passed in object in Leg and Waypoint models

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - adds .getFixPositionCoordinates() to FixCollection.

n8rzz committed:
feature/ATC-53 - Adds aircraftConstants and updates refs to use new file.

n8rzz committed:
feature/ATC-53 - Cleans and streamlines Waypoint, adds example comment to RouteModel.

n8rzz committed:
feature/ATC-53 - Adds docblocs to Leg
* Adds aircraftConstants to Leg
* Adds new method to get waypoint models for sid in AirportModel

n8rzz committed:
feature/ATC-53 - Adds restriction parsing to StandardRouteWaypointModel

n8rzz committed:
feature/ATC-53 - Adds .gemerateFmsWaypoint() to StandardWaypointModel.

n8rzz committed:
feature/ATC-53 - Updates Leg to use the RouteModel and
StandardRouteCollections when dealing with sids and stars.
* Removes inline string transformation logic from Leg

n8rzz committed:
feature/ATC-53 - Updates FixCollection.findFixByName() to uppercase the passed fixName before searching.

n8rzz committed:
feature/ATC-53 - Begins implementing sid and starCollection with RouteModel within the AircraftInstanceModel
- Adds doc blocks to Leg
- Adds new methods to Leg to reset the current waypoint array and for adding a new Waypoint to the waypoint array.

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - Updates comments and removes old TODOs

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - Abstracts restrictions logic to Waypoint
* Removes fms.clearedAsFiled() method
* Consolidates runSID() and climbViaSid() logic

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - Simplifies .descendViaSTAR()
* Adds .getStarLegWaypoints()
* Removes dead code

n8rzz committed:
feature/ATC-53 - Replaces fms ref with airportInstanceModel ref in Waypoint
* Adds tests and supporting fixutes and mocks for Waypoint

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

Conflicts:
- src/assets/scripts/canvas/CanvasController.js
- test/airport/fix/FixCollection.spec.js

n8rzz committed:
feature/ATC-53 - Adds _get to Waypoint for populating class properties, adds additional test case.

n8rzz committed:
feature/ATC-77 - Adds initial code for ModelSourceFactory and ModelSourcePool

n8rzz committed:
Merge branch 'feature/ATC-100' of https://github.com/n8rzz/atc into feature/ATC-77

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

Conflicts:
- src/assets/scripts/aircraft/AircraftFlightManagementSystem.js
- src/assets/scripts/airport/StandardRoute/StandardRouteWaypointModel.js
- test/airport/fix/FixCollection.spec.js
- test/airport/standardRoute/StandardRouteWaypointModel.spec.js

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-77

n8rzz committed:
feature/ATC-77 - Implements ModelSourcePool with FixCollection and FixModel
* Adds BaseCollection to FixCollection
* Changes name of Position to PositionModel

n8rzz committed:
feature/ATC-77 - Adds comments about future models to add to pool
* Updates tests to use implemented model, FixModel

n8rzz committed:
feature/ATC-77 - Adds documentation and TODOs

n8rzz committed:
feature/ATC-77 - updates deprecated name in failing unit test.

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - Fixes merge conflict wonkiness, several items were
merged incorrectly causing failing tests.

n8rzz committed:
feature/ATC-53 - Updates Waypoint to use ternay assignment inside of .setAltitude() and .setSpeed(). removes ava.skip from skipped tests in Waypoint.spec

n8rzz committed:
feature/ATC-53 - updates _generateManualWaypoint() to pass route not fix

n8rzz committed:
feature/ATC-53 - Adds enum for flight category
- Replaces KDGB with UNASSIGNED

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-77

n8rzz committed:
feature/ATC-53 - Uncomments ga block in index.html

n8rzz committed:
feature/ATC-53 - adds _isNil to fms

n8rzz committed:
Merge branch 'release/3.1.0' of https://github.com/n8rzz/atc into gh-pages

Conflicts:
- src/assets/scripts/airport/airportLoadList.js

n8rzz committed:
ARCH - version bump

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

erikquinn committed:
Change $("#airport") to $("#clock")

erikquinn committed:
Add UTC clock

erikquinn committed:
Refactor game clock to new class GameClockView

n8rzz committed:
feature/ATC-53 - Removes _runwayCollection from StandardRouteModel
* Refactors StandardRouteModel to use _entryCollection, _bodySegmentModel and _exitCollection for both SIDs and STARs
* Updates tests
* Adds istanbul-ignore to constructors inheriting one of the Base classes

n8rzz committed:
Merge branch 'gh-pages' of https://github.com/zlsa/atc into gh-pages

Conflicts:
- src/assets/scripts/airport/airportLoadList.js

n8rzz committed:
Merge branch 'gh-pages' of https://github.com/n8rzz/atc into release/3.2.0

n8rzz committed:
ARCH - adds bundles back to .gitignore and removes cached bundles after zlsa#gh-pages pull

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-123 - Moves gameOptions to constants dir
* Adds new GAME_OPTION_NAME enum
* Adds PARSED_COMMAND_NAME enum to InputController

n8rzz committed:
feature/ATC-144 - Adds new getter in StandardRouteCollection that
returns an object that can be used by the CanvasController to draw the
lines between SID fixes

n8rzz committed:
feature/ATC-144 - Adds documentation and TODOs to CanvasController

n8rzz committed:
ARCH - Adds v3.2.0 section to CHANGELOG.md

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-144

n8rzz committed:
feature/ATC-144 - Adds changelog entry

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-77

n8rzz committed:
feature/ATC-77 - Adds changelog entry

n8rzz committed:
bugfix//ATC-145 - Moves _comment blocks to be within object they are
describing in airport json.

n8rzz committed:
feature/ATC-77 - Updates ModelSourcePool.releaseReusable() to accept
arguments for creating a fully formed modelSource instance, eliminating
the need for the caller to also run .init() once a modelSource has been
obtained.

n8rzz committed:
Merge pull request #147 from n8rzz/bugfix/ATC-145

bugfix/ATC-145

n8rzz committed:
featureATC-144 - Fixes spelling error in StandardRouteCollection

n8rzz committed:
Merge pull request #146 from n8rzz/feature/ATC-144

feature/ATC-144

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-77

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge pull request #143 from n8rzz/feature/ATC-123

feature/ATC-123

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - Adds doc blocks to AircraftFlightManagementSystem class properties

n8rzz committed:
Merge branch 'release/3.2.0' of https://github.com/n8rzz/atc into develop

n8rzz committed:
feature/ATC-52 - Removes `$.each()` from `AirportModel` in favor of `_forEach()` and uses `_get()` inside `AircraftModel.parse()` instead of if statements

n8rzz committed:
feature/ATC-55 - Moves properties shared by all `Arrival` classes up to `ArrivalBase`

n8rzz committed:
ARCH - Fixes changelog formatting that resulted from a bad merge

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-55

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-53

n8rzz committed:
feature/ATC-53 - Updates changelog

n8rzz committed:
Merge pull request #89 from n8rzz/feature/ATC-53

feature/ATC-53

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-77

n8rzz committed:
feature/ATC-125 - moves creation of Legs and Waypoints to constants instead of as method arguments

n8rzz committed:
Merge pull request #141 from n8rzz/feature/ATC-125

feature/ATC-125

n8rzz committed:
feature/ATC-77 - Updates doc block method reference to correct method name

n8rzz committed:
Merge pull request #118 from n8rzz/feature/ATC-77

feature/ATC-77

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-52

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-55

Conflicts:
- CHANGELOG.md

n8rzz committed:
feature/ATC-128 - Moves fms files to FlightManagementSystem directory

n8rzz committed:
feature/ATC-128 - Updates changelog

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-128

n8rzz committed:
feature/ATC-17 - Moves `.parseCoordinate()` out of `PositionModel` and into `unitConverters`

n8rzz committed:
Merge pull request #152 from n8rzz/feature/ATC-52

feature/ATC-52

n8rzz committed:
Merge pull request #156 from n8rzz/feature/ATC-128

feature/ATC-128

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-55

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge pull request #153 from n8rzz/feature/ATC-55

feature/ATC-55

n8rzz committed:
feature/ATC-151 - streamlines flight number generation and adds new method to add new callsigns to the existing list.

n8rzz committed:
feature/ATC-151 - Adds .isCallsignInList() boolean helper, adds recursion for when a flight number already exists

n8rzz committed:
feature/ATC-54 - Completely removes AirportModel.sid from app
* Adds new methods to StandardRouteCollection and StandardRouteModel

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-17

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge pull request #157 from n8rzz/feature/ATC-17

feature/ATC-17

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-151

n8rzz committed:
feature/ATC-151 - Adds docblock and console.warn to AircraftController.addCallsignToList()

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-54

n8rzz committed:
feature/ATC-54- Completely removes AirportModel.stars property in favor
of AirportModel.starCollection

n8rzz committed:
feature/ATC-163 - Adds `RouteModel` to `AircraftInstanceModel.runSTAR` for easier handling of a route string

n8rzz committed:
feature/ATC-159 - Adds doc blocks to PositionModel
* Adds initial tests for PositionModel

n8rzz committed:
feature/ATC-54 - Fixes typo in StandardRouteCollection

n8rzz committed:
Merge pull request #162 from n8rzz/feature/ATC-54

feature/ATC-54

n8rzz committed:
Merge pull request #158 from n8rzz/feature/ATC-151

feature/ATC-151

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-163

n8rzz committed:
Merge pull request #165 from n8rzz/feature/ATC-163

feature/ATC-163

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-159

n8rzz committed:
feature/ATC-159 - Pulls distanceToPoint out of PositionModel
* Begins simplifying logic in PositionModel

n8rzz committed:
feature/ATC-169 - Adds Express server and basic server script.

n8rzz committed:
feature/ATC-159 - Adds documentation about Haversine Formula, adds .distEuclid() method to math/circle.js for reference.

n8rzz committed:
feature/ATC-159 - Adds static method .calculatePosition() that can be
used without instantiating a new instance
* Adds positionModelHelpers for shared logic between class and static method
* Adds REGEX to globalConstants

erikquinn committed:
Reorganize GameClockView class

n8rzz committed:
feature/ATC-159 - Integrates PositionModel.calculatePosition() into AirportModel

n8rzz committed:
Merge pull request #140 from n8rzz/feature/ATC-135

feature/ATC-135

n8rzz committed:
Merge branch 'release/3.2.0' of https://github.com/n8rzz/atc into develop

n8rzz committed:
feature/ATC-167 - Consolidates fixtures in to one root fixtures folder

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-159

n8rzz committed:
feature/ATC-159 - Updates changelog

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-169

n8rzz committed:
feature/ATC-169 - Adds Procfile

n8rzz committed:
feature/ATC-169 - Adds process.env.PORT to server file

n8rzz committed:
feature/ATC-169 - adds npm packaces to dependencies for heroku

n8rzz committed:
feature/ATC-169 - adds .travis.yml

n8rzz committed:
feature/ATC-169 - fixes typo with import line in test

n8rzz committed:
feature/ATC-169 - Fixes more typos in test imports

n8rzz committed:
Merge pull request #168 from n8rzz/feature/ATC-159

feature/ATC-159

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-167

Conflicts:
- test/fixtures/airportFixtures.js

n8rzz committed:
feature/ATC-167 - Finishes consolidating fixtures in fixtures/ directory

n8rzz committed:
Merge pull request #172 from n8rzz/feature/ATC-167

feature/ATC-167

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-169

Conflicts:
- test/airport/standardRoute/RouteSegmentCollection.spec.js
- test/airport/standardRoute/RouteSegmentModel.spec.js
- test/airport/standardRoute/StandardRouteCollection.spec.js

n8rzz committed:
feature/ATC-169 - Updates changelog and removes comments from travis.yml

n8rzz committed:
feature/ATC-114 - Adds CommandParser and CommandModel and basic tests. Adds CommandParser inside console.log to InputController

n8rzz committed:
feature/ATC-169 - Adds postinstall script back in to package.json

erikquinn committed:
Fix minor spelling error

erikquinn committed:
Fix PositionModel.calculatePosition static method

erikquinn committed:
Consolidate duplicate functions for calculating bearing to a point

erikquinn committed:
Simplify coordinate parsing process and organization

erikquinn committed:
Change `PositionModel.gps` getter to [lat,lon] order
- and replace all references to the former `PositionModel.gps` that require the [x,y] order to a new getter called `PositionModel.gpsXY`

erikquinn committed:
Simplify PositionModel._calculateScreenPosition by using static calculatePosition method

n8rzz committed:
feature/ATC-114 - Adds support for unicode character commands
* Modifies how CommandModel objects are built
* Updates tests

n8rzz committed:
feature/ATC-169 - removes branch specificity in travis.yml

erikquinn committed:
Change Position construction to only use refPos if available

n8rzz committed:
feature/ATC-114 - Begins switch to CommandParser for top level commands (excluding transmit)
* Adds InputController.processCOmmand() method and simplifies branching logic by using switch

n8rzz committed:
feature/ATC-114 - Continues CommandParser integration
* Adds new getter to CommandModel
* Adds additional test cases

erikquinn committed:
Amend PositionModel tests to pass

n8rzz committed:
feature/ATC-114 - Adds commandMap, argumentValidators and argumentParsers
* COMMAND_DEFINITION objects the define a validator and a parser to validated and parse each command
* Simplifies some of the run functions within AircraftInstanceModel

n8rzz committed:
feature/ATC-114 Adds headingValidator and headingParser
* Adds ERROR_MESSAGE enum for argumentValidators error messaging
* Adds convertToThousands() and convertStringToNumber() to unitConverters

n8rzz committed:
feature/ATC-114 - Adds WAYPOINT_NAV_MODE constants to runHeading()

n8rzz committed:
feature/ATC-114 - Finishes headingParser
* Abstracts errors to ERROR_MESSAGE
* Adds more doc blocks

n8rzz committed:
feature/ATC-114 - Adds holdValidator and tests

n8rzz committed:
feature/ATC-114 - adds holdParser and brings parser/validator tests up to 100%

n8rzz committed:
feature/ATC-114 - Adds _commandDefinition as a class property. Updates failing test for holdParser.

n8rzz committed:
feature/ATC-114 - moves .unicodeToString() and UNICODE regex to proper files

n8rzz committed:
feature/ATC-114 - Updates tests for CommandParser

erikquinn committed:
Restore terrain

erikquinn committed:
Ensure 'removed' aircraft are removed from Runway queues

erikquinn committed:
Fix wrong type fed by setDepartureRunway and add errors

n8rzz committed:
feature/ATC-114 - Adds documentation to CammandParser and CommandModel. Adds encapsulated boolean method for Parser

erikquinn committed:
Address review comments

erikquinn committed:
Fix to `fixRadialDist` function
- Restores pre-spawned aircraft

erikquinn committed:
Add test for `fixRadialDist` function

n8rzz committed:
feature/ATC-176 - Updates changelog

n8rzz committed:
Merge pull request #177 from n8rzz/feature/ATC-176

feature/ATC-176

n8rzz committed:
Merge pull request #173 from n8rzz/feature/ATC-169

feature/ATC-169

n8rzz committed:
ARCH - adds slack integration for travis.

n8rzz committed:
ARCH - adds missing express dependency

n8rzz committed:
ARCH - update node and npm versions in package.json

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-114

Conflicts:
- src/assets/scripts/constants/globalConstants.js

n8rzz committed:
ARCH - adds app.json to enable heroku review apps

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-114

n8rzz committed:
ARCH - removes backslashes in index.html urls for github pages

n8rzz committed:
feature/ATC-114 - Finishes major documentation
- Abstracts ERROR_MESSAGES to new file
- Adds _parseUserCommand() method to InputController to encapsulate actual parsing/errors of the CommandParser
- Adds validators and parsers to System commands

n8rzz committed:
feature/ATC-114 - Updates failing test

n8rzz committed:
feature/ATC-114 - Adds directionNormalizer() for headingParser
* Updates headingValidator() to accept only one or two arguments
* Updates speed.parse to use convertStringToNumber instead of convertToThousands()
* Fixes typos

n8rzz committed:
feature/ATC-114 - Updates holdParser to initialize turnDirection and legLength as null instead of empty strings

n8rzz committed:
feature/ATC-114 - Fixes broken test for holdParser

n8rzz committed:
feature/ATC-114 - Updates headingParser() to initialize direction as null instead of an empty string

n8rzz committed:
feature/ATC-114 - Splits command definitions out to smaller constants with grouped concerns.
* Adds documentation for each new constant

n8rzz committed:
feature/ATC-114 - Adds space after unicode commands in InputController

n8rzz committed:
feature/ATC-114 - Reverts runSID() method to current shape on v3.1.0

n8rzz committed:
bugfix/ATC-185 - Adds `_isNumber()` check instead of `!magneticNorth` inside `PositionModel.calculatePosition()` and the `AirspaceModel` constructor.

n8rzz committed:
feature/ATC-114 - Updates runSID() method to be a little lighter on branching logic and consolidates var declaration to the top of the method

n8rzz committed:
feature/ATC-114 - Adds processTransmitCommand method
* Simplifies .matchCallsign()

Dabea committed:
Updated function names to addItems and removeItems

Changed function names from init and destory to addItems and removeItems
this is for issue #186

n8rzz committed:
feature/ATC-114 - Adds helper method that can discern if the letter l should be used as a new command or as an argument for the heading command

n8rzz committed:
feature/ATC-114 - Adds specific user test cases to CommandParser

n8rzz committed:
feature/ATC-114 - Adds additional test case for undefined CommandModel
* Removes dead code from try/catch in InputController

n8rzz committed:
bugfix/ATC-191 - Adds failing test cases for parseElevation() bug reported under zlsa#756

n8rzz committed:
Merge pull request #187 from n8rzz/bugfix/ATC-185

bugfix/ATC-185

n8rzz committed:
feature/ATC-114 - Adds bat tick to KEY_CODES in InputController and adds additional case to handle that character

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-114

Dabea committed:
Updats the changelog with the changes made for #186

n8rzz committed:
Merge pull request #190 from n8rzz/feature/ATC-186

feature/ATC-186

panther2 commented:
I am not sure if this is the right place to post issues related to the above stated release environment
https://atc-release.herokuapp.com/

I have tested EDDM, EDDT and EDDF as well as EGLL departures:
Although clearing departing planes with
id taxi
id caf
id to
I get a "no altitude assigned" message.

It works as expected at KSFO. I did not test other airports.
Is the functionality in any way limited in the environment?
Or have I encountered a bug?

n8rzz committed:
feature/ATC-194 - Adds gulp-cli and adds SO link for windows + gulp issues

eliuuk commented:
Shouldn't cvs sort this altitude not assigned out?

On 13 Dec 2016, 21:33 +0000, P@nther <notifications@github.com>, wrote:
>
> I am not sure if this is the right place to post issues related to the above stated release environment
> https://atc-release.herokuapp.com/
>
>
> I have tested EDDM, EDDT and EDDF as well as EGLL departures:
> Although clearing departing planes with
> id taxi
> id caf
> id to
> I get a "no altitude assigned" message.
>
>
> It works as expected at KSFO. I did not test other airports.
> Is the functionality in any way limited in the environment?
> Or have I encountered a bug?
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub (https://github.com/zlsa/atc/pull/755#issuecomment-266868726), or mute the thread (https://github.com/notifications/unsubscribe-auth/ASgmWPxLTM036VfY1zb8eDZdS3jX-hDvks5rHw9GgaJpZM4LKjsM).
>
>
>

n8rzz committed:
bugfix/ATC-196 - Adds initial failing test for #196 bug

n8rzz committed:
bugfix/ATC-196 - Adds temporary fix for case when only body and rwy exist in a sid definition

n8rzz committed:
Merge pull request #195 from n8rzz/feature/ATC-194

feature/ATC-194

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-132

n8rzz committed:
feature/ATC-132 - Adds changelog entry for #132

n8rzz committed:
Merge pull request #180 from n8rzz/feature/ATC-132

feature/ATC-132

Dabea committed:
Changes routeString to RouteCode

routeCode is a better term since it is the correct term and also you do
not want to have the type evaluated in the name of variables and
functions

Dabea committed:
Changes routeString to routeCode

Dabea committed:
Merge branch 'develop' into feature/ATC-188

n8rzz committed:
Merge pull request #200 from n8rzz/feature/ATC-188

Feature/atc 188

Dabea committed:
fixes bug where parseElevation() returns NaN

Dabea committed:
Merge branch 'develop' into bugfix/ATC-191

n8rzz committed:
feature/ATC-196 - Updates test for bug condition and ensures green
- Adds console.error to notify users of error but still allow functionality

n8rzz committed:
feature/ATC-114 - removes dead code for airport system command and enforces a single argument for airport command

n8rzz committed:
feature/ATC-114 - Adds custom validator for fix command

erikquinn committed:
Create AircraftInstanceModel.isInsideAirspace method

erikquinn committed:
Only trigger collision if aircraft are inside the airspace

erikquinn committed:
Update changelog

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-114

n8rzz committed:
feature/ATC-114 - Adjusts spacing in keydown case blocks
- Adds asterisk unicode to unicode commands

n8rzz committed:
feature/ATC-114 - Adds new functions to support 1 to 3 args for hold that be received in any order
- Adjusts spacing in InputController

n8rzz committed:
feature/ATC-114 - Adjusts hold command to accept 0 to 3 arguments.
* Adds more documentation

n8rzz committed:
feature/ATC-114 - Adds missing space in InputController to subtract case

n8rzz committed:
feature/ATC-114 - Adds updates to the docs for new CommandParser

n8rzz committed:
Merge pull request #179 from n8rzz/feature/ATC-114

feature/ATC-114

n8rzz committed:
ARCH - adds missing changelog entry for #114

n8rzz committed:
bugfix/ATC-196 - Updates console.error messaging

n8rzz committed:
feature/ATC-216 - Removes pegjs parser and dependencies

n8rzz committed:
feature/ATC-211 - Updates escape key handling to clear everything but callsign if commands are present

n8rzz committed:
ARCH - adds updates to altitude section about the expedite argument

Dabea committed:
add link to original issue #756

Commit:
add link to original issue #756

Dabea committed:
Combined If statements and added comments

Dabea committed:
Combined If statements and added comments

Dabea committed:
fixed parentheses placement

Dabea committed:
fixed parentheses placement

Dabea committed:
Merge branch 'bugfix/ATC-191' of https://github.com/n8rzz/atc into bugfix/ATC-191

n8rzz committed:
Merge pull request #201 from n8rzz/bugfix/ATC-191

bugfix/ATC-191

erikquinn committed:
Merge develop

erikquinn committed:
fix improper use of _.has / _.isNil

erikquinn committed:
Rename and repair method for weight-on-wheels logic from .wow() to .isOnGround()

erikquinn committed:
Replace all references to .wow() with .isOnGround()

n8rzz committed:
Merge pull request #217 from n8rzz/feature/ATC-216

feature/ATC-216

n8rzz committed:
Merge pull request #218 from n8rzz/feature/ATC-211

feature/ATC-211

erikquinn committed:
Clean up AircraftInstanceModel.isOnGround() method

n8rzz committed:
Merge branch 'release/3.2.0' of https://github.com/n8rzz/atc into bugfix/ATC-196

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge pull request #198 from n8rzz/bugfix/ATC-196

bugfix/ATC-196

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-134

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge branch 'feature/ATC-134' of https://github.com/n8rzz/atc into feature/ATC-134

n8rzz committed:
Merge pull request #209 from n8rzz/feature/ATC-134

feature/ATC-134

n8rzz committed:
feature/ATC-175 - Adds tests for new throws in StandardRouteModel, updates changelog

n8rzz committed:
Merge branch 'develop' of https://github.com/n8rzz/atc into feature/ATC-175

Conflicts:
- CHANGELOG.md

n8rzz committed:
Merge pull request #226 from n8rzz/feature/ATC-175

feature/ATC-175

n8rzz committed:
Merge branch 'release/3.2.0' of https://github.com/n8rzz/atc into release/3.2.0

n8rzz commented:
@erikquinn everything merged fine locally and tests pass.  :shipit:

n8rzz committed:
ARCH - updates contributors list
